beetsplug/deezer.py

```diff
@@ -63,6 +63,22 @@ def func(lib, opts, args):
 
         return [deezer_update_cmd]
 
+    def fetch_data(self, url):
+        try:
+            response = requests.get(url, timeout=10)
+            response.raise_for_status()
+            data = response.json()
+        except (
+            requests.ConnectionError,
+            requests.exceptions.RequestException,
```

Comment:
ah and now found this older comment in another place, so actually this applies here as well: https://github.com/beetbox/beets/pull/4984#discussion_r1402470534

Reply:
hmm @sampsyo didn't you mean the other way round in that comment? If we catch RequestException we'd also catch ConnectionError?

Reply:
but hey @arsaboo I wanted to add: Just my thoughts and I don't really know if catching _all_ exceptions `requests` offers is the perfect idea! Maybe @sampsyo _did_ mean it like he wrote in the first place (ConnectionError catching is enough).

The rule of thumb for catching errors often is: Don't catch everything.

Maybe there are situations where we actually want the importer to crash and tell the beets user that there is a serious problem. If we catch everything then in a normal important run without seeing logs the importer would for example simply fail silently and the user might not even see the reason why they are not getting any results from deezer!!! just think about that and look at this page again, it shows closely which errors derive from which: https://requests.readthedocs.io/en/stable/api/#exceptions

Reply:
Often these are not serious problems....the API errors out or something trivial (and often resolves in the next run). In any case, I will wait for a consensus before making changes. 

Also, we are not masking anything....we are telling the user that there was an error. Instead of `self._log.debug`, we can make it `self._log.error` so that it is not missed.

Reply:
ah yeah that's true! making it error-level actually shows it always! yeah good point!

Reply:
cool...updated the PR.